enrichments/trace/internal/elastic/span.go

```diff
@@ -300,6 +303,40 @@ func (s *spanEnrichmentContext) setServiceTarget(span ptrace.Span) {
 	}
 }
 
+func (s *spanEnrichmentContext) setDestinationService(span ptrace.Span) {
+	var destnResource string
+	if s.peerService != "" {
+		destnResource = s.peerService
+	}
+
+	switch {
+	case s.isDB:
+		if destnResource == "" && s.dbType != "" {
+			destnResource = s.dbType
+		}
+	case s.isMessaging:
+		if destnResource == "" && s.messagingSystem != "" {
+			destnResource = s.messagingSystem
+		}
+		// For parity with apm-data, destn resource does not handle
+		// temporary destination flag. However, it is handled by
+		// service.target fields and we might want to do the same here.
+		if destnResource != "" && s.messagingDestinationName != "" {
+			destnResource += "/" + s.messagingDestinationName
+		}
+	case s.isRPC, s.isHTTP:
+		if destnResource == "" {
+			if res := getHostPort(s.urlFull, s.urlDomain, s.urlPort); res != "" {
+				destnResource = res
+			}
```

Comment:
In apm-data we omit the scheme default port for HTTP. i.e. omit 80 for http://, and 443 for https://. Should we have that here too? Or is the expectation that those will not be present in `url.full`?

Reply:
Ah, thanks for raising this. I forgot to add the explanation for this. The OTel schema describes `url.full` to follow [RFC3986](https://www.rfc-editor.org/rfc/rfc3986)([ref](https://opentelemetry.io/docs/specs/semconv/attributes-registry/url/)). As per this RFC:

>    A scheme may define a default port.  For example, the "http" scheme
   defines a default port of "80", corresponding to its reserved TCP
   port number.  The type of port designated by the port number (e.g.,
   TCP, UDP, SCTP) is defined by the URI scheme.  URI producers and
   normalizers should omit the port component and its ":" delimiter if
   port is empty **or if its value would be the same as that of the
   scheme's default**.

([Link to the quote](https://www.rfc-editor.org/rfc/rfc3986#section-3.2.3))

So, we should expect `url.full` without the default ports.

Reply:
Nice, thanks.